Mathlib/FieldTheory/Finite/Basic.lean

```diff
@@ -201,6 +228,84 @@ theorem sum_pow_units [DecidableEq K] (i : ℕ) :
         apply forall_congr'; intro x; simp [Units.ext_iff]
 #align finite_field.sum_pow_units FiniteField.sum_pow_units
 
+/-- The sum of a nontrivial subgroup of the units of a field is zero. -/
+theorem sum_subgroup_units_zero_of_ne_bot {G : Subgroup Kˣ} [Fintype G] (hg : G ≠ ⊥) :
+    ∑ x : G, (x.val : K) = 0 := by
+  rw [Subgroup.ne_bot_iff_exists_ne_one] at hg
+  rcases hg with ⟨a, ha⟩
+  -- The action of a on G as an embedding
+  have hinj := mul_right_injective a
+  let a_mul_emb : G ↪ G := ⟨(a * ·), hinj⟩
+  -- let a_mul_emb : G ↪ G := by exact mulLeftEmbedding a -- TODO factor this? In library?
+  -- ... and leaves G unchanged
+  have h_unchanged : Finset.univ.map a_mul_emb = Finset.univ := by simp
+  -- Therefore the sum of x over a G is the sum of a x over G
+  have h_sum_map := Finset.univ.sum_map a_mul_emb fun x => ((x : Kˣ) : K)
+  -- ... and the former is the sum of x over G.
+  -- By algebraic manipulation, we have Σ G, x = ∑ G, a x = a ∑ G, x
+  rw [h_unchanged, Function.Embedding.coeFn_mk] at h_sum_map
+  simp_rw [Subgroup.coe_mul, Units.val_mul, ← Finset.mul_sum] at h_sum_map
+  -- thus one of (a - 1) or ∑ G, x is zero
+  have hzero : (((a : Kˣ) : K) - 1) = 0 ∨ ∑ x : ↥G, ((x : Kˣ) : K) = 0 := by
+    rw [←mul_eq_zero, sub_mul, ← h_sum_map, one_mul, sub_self]
+  rcases hzero with h | h
+  · -- If the former, we reach contradiction from a ≠ 1
+    exfalso
+    apply ha
+    ext
+    rw [←sub_eq_zero]
+    exact h
+  · -- If the latter, we are done
+    exact h
+
+/-- The sum of a subgroup of the units of a field is 1 if the subgroup is trivial and 1 otherwise -/
+theorem sum_subgroup_units {G : Subgroup Kˣ} [Fintype G] [Decidable (G = ⊥)] :
```

Comment:
```suggestion
theorem sum_subgroup_units {G : Subgroup Kˣ} [Fintype G] :
```

and use `classical` instead in the proof, or even `rcases eq_or_ne G \bot with rfl | G_bot`

Reply:
It is impossible to state the type of the conclusion without this

Reply:
Note that the above lemma has a decidability assumption as well. Happy to get more guidance on how to remove this if possible.

Reply:
Oh, d'oh, yes the ITE depends on it!